lib/backend/dbi.c

```diff
@@ -77,7 +78,7 @@ dbDetectBackend(rpmdb rdb)
 	}
     }
 
-    if (!cfg) {
+    if (!cfg && ((rdb->db_mode & O_ACCMODE) != O_RDONLY || (rdb->db_flags & RPMDB_FLAG_REBUILD) != 0)) {
```

Comment:
Hmm, does this actually need the REBUILD flag test at all, with the O_RDONLY check in place?

Reply:
Maybe not for rpm itself, but I think it should stay to guard against bad librpm users. Otherwise it'll crash with a null pointer deref below in the REBUILD case when trying to print the warning message.

Reply:
But that's no longer the case in this latest version, right? 
There are quite a few important and subtle checks here so I think we shouldn't mix it up with any redundant ones.

Reply:
It will no longer crash, but it's still wrong. IMHO it's better to error out early when opening the old database in rebuilddb mode instead of later when opening the new database.

Reply:
Okay, fair enough.